pkg/ctl/install/install.go

```diff
@@ -0,0 +1,435 @@
+	}
+	kubernetesClientConfigs, err := ctl.NewClient(cfg)
+	if err != nil {
+		return nil, err
+	}
+	k8sConfig := kubernetesClientConfigs.Config
+
+	k8sRestConfig, err := clientcmd.NewDefaultClientConfig(*k8sConfig, &clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		return nil, errors.Errorf("cannot create Kubernetes client configuration: %s", err)
+	}
+	k8sClientSet, err := kubernetes.NewForConfig(k8sRestConfig)
+	if err != nil {
+		return nil, errors.Errorf("cannot create Kubernetes client set: %s", err)
+	}
+	fi := &fluxInstaller{
+		opts: opts,
+		resourceCmd: rc,
+		k8sConfig: k8sConfig,
+		k8sRestConfig: k8sRestConfig,
+		k8sClientSet: k8sClientSet,
+	}
+	return fi, nil
+}
+
+
+func (fi *fluxInstaller) run(ctx context.Context) error {
+	manifests, err := getFluxManifests(fi.opts.templateParams, fi.k8sClientSet)
+	if err != nil {
+		return fmt.Errorf("failed to create flux manifests: %s", err)
+	}
+
+
+	logger.Info("Cloning %s", fi.opts.templateParams.GitURL)
+	cloneDir, err := fi.cloneRepo(ctx)
+	if err != nil {
+		return fmt.Errorf("cannot clone repository %s: %s", fi.opts.templateParams.GitURL, err)
+	}
+	cleanCloneDir := false
+	defer func() {
+		if cleanCloneDir {
+			os.RemoveAll(cloneDir)
+		} else {
+			logger.Critical("You may find the local clone of %s used by eksctl at %s",
+				fi.opts.templateParams.GitURL,
+				cloneDir)
+		}
+	}()
+
+	logger.Info("Writing Flux manifests")
+	fluxManifestDir := filepath.Join(cloneDir, fi.opts.gitFluxPath)
+	if err := writeFluxManifests(fluxManifestDir, manifests); err != nil {
+		return err
+	}
+
+	if fi.opts.amend {
+		logger.Info("Stopping to amend the the Flux manifests, please exit the shell when done.")
+		if err := runShell(fluxManifestDir); err != nil {
+			return err
+		}
+	}
+
+	// If we need to create a Namespace, we do it synchronously before invoking kubectl apply.
+	// Otherwise there is a race between the creation of the namespace and the other resources living in it
+	if _, ok := manifests[namespaceFileName]; ok {
+		logger.Info("Creating namespace %s", fi.opts.templateParams.Namespace)
+		if err:= createNamespaceSynchronously(fi.k8sClientSet, fi.opts.templateParams.Namespace); err != nil {
+			return err
+		}
+	}
+
+	logger.Info("Installing Flux into the cluster")
+	if err := fi.applyManifestsInDir(ctx, fluxManifestDir); err != nil {
+		return err
+	}
+
+	logger.Info("Waiting for Flux to start")
+	fluxSSHKey, err :=  waitForFluxToStart(ctx, fi.opts, fi.k8sRestConfig, fi.k8sClientSet)
+	if err != nil {
+		return err
+	}
+	logger.Info("Flux started successfully")
+
+	logger.Info("Committing and pushing Flux manifests to %s", fi.opts.templateParams.GitURL)
+	if err := fi.addCommitAndPushFluxManifests(ctx, cloneDir); err != nil {
+		return err
+	}
+	cleanCloneDir = true
+
+	logger.Info("In order for Flux to operate properly make sure it has SSH access to %s",
+		fi.opts.templateParams.GitURL)
+	logger.Info("Flux's current Public SSH key is:\n%s", fluxSSHKey.Key)
```

Comment:
"Flux's" reads weirdly, let's just construct this differently, also capital 'P' is not needed